pyanaconda/modules/storage/bootloader/systemd.py

```diff
@@ -0,0 +1,192 @@
+
+        if len(devices) == 0:
+            return
+
+    # copy console update from grub2.py
+    def write_config_console(self, config):
+        log.info("systemd.py: write_config_console")
+        if not self.console:
+            return
+
+        console_arg = "console=%s" % self.console
+        if self.console_options:
+            console_arg += ",%s" % self.console_options
+        self.boot_args.add(console_arg)
+
+
+
+    def write_config(self):
+        log.info("systemd.py: write_config systemd start")
+        # So at this point, all this is pretty much redundant
+        self.write_config_console(None)
+
+        # just rewrite the loader.conf
+        #cfg_file = config_file(self)
+        config_path = "%s%s" % (conf.target.system_root, "/boot/efi/loader/loader.conf")
+        log.info("systemd.py: write_config systemd loader conf : %s ", config_path)
+
+        # outside of resetting the default, this file should probably be moved to
+        # stubby as well?
+        config = open(config_path, "w")
+        config.write("timeout 3\n")
+        # create a default entry? No need right now..
+        # config.write("default 4142287a09f64398a50ac387bf9e1fca-*\n")
+        config.write("#console-mode keep\n")
+        config.close()
+
+        # do the bls command line here?
+        # update /etc/kernel/cmdline
+        # should look something like "root=UUID=45b931b7-592a-46dc-9c33-d38d5901ec29 ro  resume=/dev/sda3"
+        config_path = "%s%s" % (conf.target.system_root, "/etc/kernel/cmdline")
+        log.info("systemd.py: write_config systemd commandline : %s ", config_path)
+        config = open(config_path, "w")
+        args = " ".join(list(self.boot_args._arguments))
+        log.info("systemd.py: systemd used boot args: %s ", args)
+        # should run this through blkid to get UUID... but what is going on with the above args...
+        args += " root=/dev/mapper/fedora_fedora-root"
+        config.write(args)
+        config.close()
+        # rather than creating a mess in python lets just
+        # write the options above, and run a script which will merge the
+        # boot cmdline (after stripping inst. and BOOT_) with the anaconda settings
+        rc = util.execWithRedirect(
+            "/usr/sbin/updateloaderentries.sh",
+            [" "],
+            root=conf.target.system_root
+
+        )
+        if rc:
+            raise BootLoaderError("failed to write boot loader configuration")
+    #
+    # installation
+    #
+
+    def installbootmgr(self, args=None):
+        log.info("systemd.py: install")
+        if args is None:
+            args = []
+
+        log.info("systemd.py: install systemd boot install (root=%s)", conf.target.system_root)
+
+        rc = util.execWithRedirect("bootctl", [ "install", "--esp-path=/boot/efi",
+                                                "--efi-boot-option-description=fedora",
+                                                "--loader-path=/EFI/fedora/" ],
```

Comment:
ditto

Reply:
I've got a local change for this at the moment, but its not clear what the upstream change will be, so I will likely drop the loader-path parts from v2, under the assumption it can be added back when upstream systemd accepts a patch.

Reply:
That discussion seems to be here: https://github.com/systemd/systemd/pull/24828 now that i've removed the arm-specific kernel-install bits, which should be redundant with the compressed EFI/kernel images.